### PR TITLE
*: remove dependency on string.h

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: include/frg/string.hpp [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+namespace {
+template<typename CharT>
+auto generic_strlen(const CharT *c) {
+	std::size_t len = 0;
+	while (*(c++)) {
+		len++;
+	}
+	return len;
+}
+}
```
Comment (avdgrinten) [inline include/frg/string.hpp:20]: Put this into `namespace frg`.

File: include/frg/string.hpp [hunk 2]
```diff
@@ -308,8 +315,8 @@ class basic_string {
 	}
 
 	int compare(const char *other) const {
-		if(_length != strlen(other))
-			return _length < strlen(other) ? -1 : 1;
+		if(_length != generic_strlen(other))
+			return _length < generic_strlen(other) ? -1 : 1;
```
Comment (avdgrinten) [inline include/frg/string.hpp:319]: We don't have to compute `generic_strlen` twice here. (Yes, that was already done before, but it's a 3 line fix).

File: include/frg/string_stub.hpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#if defined(__has_include) && __has_include(<string.h>)
```
Comment (avdgrinten) [inline include/frg/string_stub.hpp:3]: Do we need `defined(__has_include)`? All supported compilers already support `__has_include`.

File: include/frg/string.hpp (hunk 2 above)
Comment (ArsenArsen) [reply include/frg/string.hpp:319]: oh, I didn't catch that since I mechanically replaced `strlen` with `generic_&`. will fix

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Review by qookei — APPROVED** [review]

**Pull request merged**